defi/src/getChains.ts

```diff
@@ -6,54 +6,61 @@ import { IChain, } from "./types";
 import { importAdapter } from "./utils/imports/importAdapter";
 import { excludeProtocolInCharts, isExcludedFromChainTvl } from "./utils/excludeProtocols";
 
-async function _checkModuleDoubleCounted(protocol: Protocol){
+async function _checkModuleDoubleCounted(protocol: Protocol) {
   const module = await importAdapter(protocol);
   return module.doublecounted
 }
 
-async function _getLastHourlyRecord(protocol: Protocol){
+async function _getLastHourlyRecord(protocol: Protocol) {
   return getLastRecord(hourlyTvl(protocol.id))
 }
 
 export async function craftChainsResponse(excludeDoublecountedAndLSD = false, useNewChainNames = false, {
   checkModuleDoubleCounted = _checkModuleDoubleCounted,
   getLastHourlyRecord = _getLastHourlyRecord,
   protocolList = protocols,
-} = {}){
-  const chainTvls = {} as {[chain:string]:number}
+} = {}) {
+  const chainTvls = {} as { [chain: string]: { tvl: number, defillama_id: string } }
   await Promise.all(
     protocolList.map(async (protocol) => {
-      if(excludeProtocolInCharts(protocol) || isExcludedFromChainTvl(protocol.category)){
+      if (excludeProtocolInCharts(protocol) || isExcludedFromChainTvl(protocol.category)) {
         return undefined;
       }
       const lastTvl = await getLastHourlyRecord(protocol)
-      if(lastTvl === undefined){
-          return
+      if (lastTvl === undefined) {
+        return
       }
-      const excludeTvl = excludeDoublecountedAndLSD && (protocol.category === "Liquid Staking" || isDoubleCounted(await checkModuleDoubleCounted(protocol), protocol.category)  === true)
+      const excludeTvl = excludeDoublecountedAndLSD && (protocol.category === "Liquid Staking" || isDoubleCounted(await checkModuleDoubleCounted(protocol), protocol.category) === true)
       if (excludeTvl) return;
       let chainsAdded = 0
-      Object.entries(lastTvl).forEach(([chain, chainTvl])=>{
-          const chainName = getChainDisplayName(chain, useNewChainNames)
-          if(chainCoingeckoIds[chainName] === undefined){
-              return
-          }
-          chainTvls[chainName] = (chainTvls[chainName] ?? 0) + chainTvl
-          chainsAdded += 1;
+      Object.entries(lastTvl).forEach(([chain, chainTvl]) => {
+        const chainName = getChainDisplayName(chain, useNewChainNames)
+        if (chainCoingeckoIds[chainName] === undefined) {
+          return
+        }
+        if (!chainTvls[chainName]) {
+          chainTvls[chainName] = { tvl: 0, defillama_id: chain }
+        }
+        chainTvls[chainName].tvl += chainTvl as number
+        chainsAdded += 1;
       })
-      if(chainsAdded === 0){ // for fetch adapters
+      if (chainsAdded === 0) { // for fetch adapters
         const chainName = protocol.chain
-        chainTvls[chainName] = (chainTvls[chainName] ?? 0) + lastTvl.tvl
+        if (!chainTvls[chainName]) {
+          chainTvls[chainName] = { tvl: 0, defillama_id: chainName }
```

Comment:
this is wrong because this will show the formatted chain name instead of the raw defillama id